src/kvstore/kvstore_dist.h

```diff
@@ -329,18 +330,21 @@ class KVStoreDist : public KVStoreLocal {
         }
         CopyFromTo(merged, &comm_buf);
       }
-
+      int dtype = merged.dtype();
+      int num_bytes = mshadow::mshadow_sizeof(dtype);
       // push to servers
       if (storage_type == kDefaultStorage) {
         if (gradient_compression_->get_type() == CompressionType::kNone) {
-          PSKV& pskv = EncodeDefaultKey(key, comm_buf.shape().Size(), true);
+          PSKV& pskv = EncodeDefaultKey(key, comm_buf.shape().Size(), true, num_bytes);
           PushDefault(key, comm_buf, pskv, priority);
         } else {
+          CHECK_EQ(dtype, mshadow::kFloat32) << "Gradient compression is only supported for "
```

Comment:
Isn't it inconsistent to say Float16 is supported while actually not all features support it?

Reply:
This is the only feature left. It needs changes to some core gradient compression methods, which is coming in a later PR. Besides all changes *in kvstore* for gradient compression to support fp16 are there already.

Reply:
Ah, sounds reasonable. Thanks!